### PR TITLE
Update User API developer docs to reflect PBKDF2 hashing algo changes

### PR DESCRIPTION
**Pull request by susanharper-okta**
…F2 hashing algo changes

## Description:
- **What's changed?** Update User API developer docs to reflect PBKDF2 hashing algo changes
- **Is this PR related to a Monolith release?** Yes 2022.12.0

## HTML View of the changes done
Hashed Password object section changes 

Screen Shot 2022-11-29 at 7 39 51 PM

Request Body 
Screen Shot 2022-11-29 at 7 40 05 PM
 

### Resolves:

* [OKTA-553281](https://oktainc.atlassian.net/browse/OKTA-553281)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md [hunk 1]
```diff
@@ -4110,17 +4110,20 @@ The password specified in the value property must meet the default password poli
...
 | salt       | String   | Only required for salted hashes. For BCRYPT, this specifies the radix64-encoded salt used to generate the hash, which must be 22 characters long. For other salted hashes, this specifies the base64-encoded salt used to generate the hash. |
 | workFactor | Number  | Governs the strength of the hash and the time required to compute it. Only required for BCRYPT algorithm. Minimum value is 1, and maximum is 20. |
 | saltOrder  | String   | Specifies whether salt was pre- or postfixed to the password before hashing. Only required for salted algorithms. |
+| iterationCount  | Number   | The number of iterations used when hashing passwords using PBKDF2. Must be >= 4096. Only required for PBKDF2 algorithm. |
+| keySize  | Number   | Size of the derived key in bytes. Only required for PBKDF2 algorithm. |
+| digestAlgorithm  | String   | Algorithm used to generate the key. Currently we support "SHA256_HMAC" and "SHA512_HMAC“. Only required for PBKDF2 algorithm. |
```
Comment (edjohns-okta) [inline packages/@okta/vuepress-site/docs/reference/api/users/index.md:4126]: Nit: I would clarify the text to indicate that iterationCount/keysize/digestAlgorithm are only to be included if the algo is PBKDF2.
Maybe something like `Specify only for PBKDF2 algorithm`

**Review by edjohns-okta — COMMENTED** [review]
Had one minor comment

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md (hunk 1 above)
Comment (hassanmahmood-okta) [reply packages/@okta/vuepress-site/docs/reference/api/users/index.md:4126]: it is there at the end of each description "Only required for PBKDF2 algorithm."

Comment (edjohns-okta) [reply packages/@okta/vuepress-site/docs/reference/api/users/index.md:4126]: OK, I'll give you a ship.  I don't want to belabor the point, which was that "Only required for PBKDF2" could be interpreted as "Only required for PBKDF2, optional for others".

**Review by edjohns-okta — APPROVED** [review]
🚢

**Comment by brianduffield-okta** [comment]
Merging to release branch

**Pull request merged**